StatusWidget/StatusWidget.swift

```diff
@@ -0,0 +1,264 @@
+//
+//  SmallStatusWidget.swift
+//  Loop
+//
+//  Created by Noah Brauner on 8/15/22.
+//  Copyright © 2022 LoopKit Authors. All rights reserved.
+//
+
+import WidgetKit
+import SwiftUI
+import LoopKit
+import LoopCore
+import HealthKit
+import CoreData
+
+class StatusWidgetProvider: TimelineProvider {
+    lazy var defaults = UserDefaults.appGroup
+    
+    lazy var healthStore = HKHealthStore()
+
+    lazy var cacheStore = PersistenceController.controllerInAppGroupDirectory()
+
+    lazy var localCacheDuration = Bundle.main.localCacheDuration
+
+    lazy var settingsStore: SettingsStore =  SettingsStore(
+        store: cacheStore,
+        expireAfter: localCacheDuration)
+
+    lazy var glucoseStore = GlucoseStore(
+        healthStore: healthStore,
+        observeHealthKitSamplesFromOtherApps: FeatureFlags.observeHealthKitGlucoseSamplesFromOtherApps,
+        storeSamplesToHealthKit: false,
+        cacheStore: cacheStore,
+        observationEnabled: false,
+        provenanceIdentifier: HKSource.default().bundleIdentifier
+    )
+
+    func placeholder(in context: Context) -> StatusWidgetEntry {
+        return StatusWidgetEntry(date: Date(), originalDate: Date(), lastLoopCompleted: nil, closeLoop: true, currentGlucose: nil, previousGlucose: nil, unit: .milligramsPerDeciliter, sensor: nil, netBasal: nil, eventualGlucose: nil)
+    }
+
+    func getSnapshot(in context: Context, completion: @escaping (StatusWidgetEntry) -> ()) {
+        update { newEntry in
```

Comment:
If `context.isPreview` is `true`, and we don't have current data from Loop, then we should display mock data, so that we give a good example of the widget for the Widget Gallery.